### DISCUSSION TITLE
Hayward Super Pump VS 700

### DISCUSSION THREAD

**Discussion by Crewski** [tagyoureit/nodejs-poolController#466] [Nixie] [answered]
I have a Hayward Super Pump VS 700 that I'm having issues with.  I have is setup via RS485 (19200 baud) and can send the speed commands and the pump responds.  However the response from the pump isn't working.  I've attached a replay.  It looks like the response from the pump is trying to be process as a chlorinator and not a Hayward VS.  I'm on commit d7b0993.

Also, are we able to update VS pump speeds via MQTT?  I looked and it doesn't seem to be listening to the config/pump topic.
[replay (5).zip](https://github.com/tagyoureit/nodejs-poolController/files/8792312/replay.5.zip)
.

**Comment by rstrouse** [comment]
Actually I see chlorinator packets in there but there is no response from the Hayward pump.  How old is your pump.  We had another person that attempted to connect via RS485 only to find out that the old VS700s do not talk back.
https://github.com/tagyoureit/nodejs-poolController/issues/393

The speeds are set by turning on and off specific circuits that have speeds assigned to them.  For instance, when you turn on solar you may want to ramp up the speed but if only the pool is running you then the speed is lower.  The circuit with the highest speed always wins.

![image](https://user-images.githubusercontent.com/47839015/170847504-a512ab19-0d62-45b6-ac55-d49fdb7c355a.png)

**Reply by Crewski** [reply]
The pump is less than a year old....just got the pool installed Sept 2021.  I've never played with RS485 before, but I'm assuming since I'm able to control my pump that my wiring is correct.  Any suggestions?

Thanks for the explanation on the pumps, that helps a lot.

**Comment by Crewski** [comment]
I do have two separate RS485 ports since the pump and chlorinator run at different BAUD rates.  I have the aquarite swg on port 1, and the hayward vs on port 2.

For port 1 I get:
```
[5/28/2022, 19:02:03] info: {"port":1,"id":1150,"valid":true,"dir":"in","proto":"chlorinator","pkt":[[],[],[16,2,0,1],[0,0],[19,16,3]],"ts": "2022-05-28T19:02:03.435-0400"}
[5/28/2022, 19:02:03] info: {"port":1,"id":1151,"valid":true,"dir":"out","proto":"chlorinator","pkt":[[],[], [16,2,80,17], [0],[115,16,3]],"ts":"2022-05-28T19:02:03.739-0400"}
```

For port 2 I'm seeing.  Seems odd that the protocol for the "in" is chlorinator. :
```
[5/28/2022, 19:02:02] info: {"port":2,"id":1147,"valid":true,"dir":"out","proto":"hayward","pkt":[[],[], [16,2,12,1,0], [78],[0,109,16,3]],"ts":"2022-05-28T19:02:02.982-0400"}
[5/28/2022, 19:02:03] info: {"port":2,"id":1148,"valid":true,"dir":"in","proto":"chlorinator","pkt":[[],[],[16,2,0,12],[0,0,78,5,118,0],[231,16,3]],"ts": "2022-05-28T19:02:03.001-0400"}
```

**Reply by rstrouse** [reply]
Oh wow you are correct.  I stared at that until it struck me that this is our missing message.   We simply have matched the wrong protocol for that return.  That message is actually supposed to be parsed like the following.  

`[[],[],[16,2,0,12,0][0,78,5,118,][0,231,16,3]]`

It just so happens that the bytes pass the CRC check or we would not have picked this up so I am wondering if the other Hayward actually was returning but it looked like a failed chlorinator packet.  I'll take a look at this and rework the header matching for this.

**Comment by rstrouse** [comment, marked as answer]
Pull njsPC and dashPanel. I added some code to route the messages properly.  After you update can I get another replay capture?

**Reply by Crewski** [reply]
I updated everything.  It looks like it is being process as "hayward" now, but I'm still not getting the pump details....dashpanel is just blank for speed and watts.  I tried it with both 1 and 2 stop bits as I think I read somewhere it used 2, but it didn't make a difference.

```
[5/30/2022, 06:35:43] info: {"port":2,"id":401,"valid":true,"dir":"out","proto":"hayward","pkt":[[],[], [16,2,12,1,0], [41],[0,72,16,3]],"ts":"2022-05-30T06:35:43.163-0400"}
[5/30/2022, 06:35:43] info: {"port":2,"id":402,"valid":true,"dir":"in","proto":"hayward","pkt":[[],[],[16,2,0,12],[0,0,41,0,135],[0,206,16,3]],"ts": "2022-05-30T06:35:43.184-0400"}
```
[replay (7).zip](https://github.com/tagyoureit/nodejs-poolController/files/8797670/replay.7.zip)

**Reply by Crewski** [reply]
I don't know if this is related or now, but I've lost communication with my Aquarite T9 SWG.  It was working in my first replay that started this discussion, and I haven't changed anything since then. The unit has the "Remotely Controller" light lit up, and the receive light on my RS485 hat flashes when the transmit is sent.

**Reply by rstrouse** [reply]
Yes this was related but the primary issue was that my test rig had a bad RS485 port so I misinterpreted the incoming bytes and ended up filtering out the SWG messages.  Pull njsPC.

**Reply by Crewski** [reply]
The SWG is properly communicating again.  The data coming back from the Hayward VS is still isn't giving me the pump details.

```
{"port":2,"id":12431,"valid":true,"dir":"out","proto":"hayward","pkt":[[],[], [16,2,12,1,0], [78],[0,109,16,3]],"ts":"2022-05-30T19:21:25.233-0400"}
{"port":2,"id":12433,"valid":true,"dir":"in","proto":"hayward","pkt":[[],[],[16,2,0,12,0],[0,78,5,100],[0,213,16,3]],"ts": "2022-05-30T19:21:25.252-0400"}
```

**Reply by rstrouse** [reply]
I posted a fix a couple of minutes ago that fixes the addresses for the pumps.  This should get the returns to now display.  Please pull and try again.  I unfortunately do not have one of these pumps so I have to examine the returns and another user sent me an output that was on another address.

**Reply by Crewski** [reply]
Awesome....looks like all is well now.  The only issue I see is that the indicator isn't lit up when the pump is running.  Beside that everything is great.  Thanks for the help.

![image](https://user-images.githubusercontent.com/8012153/171072459-0f8c76d4-f9cd-4265-9f71-43f6fedc83d5.png)

**Reply by rstrouse** [reply]
I'll have a look.  This pump doesn't give us a drive state so it is possible that we may need to make one up.

**Reply by rstrouse** [reply]
Pull dashPanel and njsPC the command is being set now.

**Reply by Crewski** [reply]
Everything seems to be good now.  Thanks!

**Comment by Crewski** [comment]
Any idea why my pump would max out at 85%.  When I try to turn it on max speed and send 100%, it'll only run at 85% (2933 RPM).  What's really strange is that if I set it at 86% it'll run at 86%, but anything over that it drops to 85%.  Replay for completeness.  Thanks.  When using the on-pump controller prior to unhooking them for RS485 it would run at 100%.
[replay (12).zip](https://github.com/tagyoureit/nodejs-poolController/files/8837606/replay.12.zip)
.

**Comment by rstrouse** [comment]
This has to be a setting on the pump.  In looking at the messages Nixie is requesting 100% and the pump is returning 85%.  There is a Max Allowed speed on the pump in the manual but I don't know how to change it.  The good news is that I know the pump can run at 100% given replays from @nate-marshall.

Here is a message sequence that should be occurring
`[16, 2, 12, 1, 1][100][0, 132, 16, 3]  // Nixie requests 100%`  
`[16, 2, 0, 12, 1][0, 100, 21, 97][0, 249, 16, 3] // Pump returns 100% and 1,561watts`

And here is what we are getting from your pump
`[16, 2, 12, 1, 0][100][0, 131, 16, 3] // Nixie requests 100%`
`[16, 2, 0, 12, 0][0, 85, 7, 25][0, 147, 16, 3] // Pump returns 85% and 719watts`

Can I get you to go through the manual for your pump and make sure you do not have a max speed set on it.   Also, if there is a priming setting it may not allow the pump to run at full speed until the priming cycle has completed.

**Reply by Crewski** [reply]
First rule of "IT", turn it off and back on.  After flipping the breaker on my pump and turning it back on, it seems to be properly responding to speeds above 85%.  100% is now returning the proper 3450 RPM.

I did check the manual and didn't see anything about a max speed.

**Reply by rstrouse** [reply]
It is definitely in the menus but it is probably only clear on how it is set with the pump in front of you.  It is also likely that you cleared that setting out of the memory on the drive by shutting it down.   It appears this setting is not for when you are connected to automation.
